src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -531,6 +537,37 @@ public void delete(String modelId, ActionListener<DeleteModelResponse> listener)
                     return;
                 }
 
+                // Get index mappings to check if model is being used by any indices
+                getMappings(modelId, getMappingsStep);
+            }, listener::onFailure);
+
+            getMappingsStep.whenComplete(getMappingsResponse -> {
+                // The mapping metadata string will always have the model id followed by `,` and another parameter, or the end of the object
+                // `}`
+                // This check is to prevent an incorrect thrown exception when one model starts with the another model's id
+                // Ex: test-model and test-model1
+                String modelIdStringInMapping1 = String.format("model_id=%s,", modelId);
+                String modelIdStringInMapping2 = String.format("model_id=%s}", modelId);
```

Comment:
@naveentatikonda the difference is that it would be searching if the string contains `test-model,` or `test-model}` instead of just `test-model`.  I'll attempt the casting solution today.